### DISCUSSION TITLE
Latex disabled since updating to v1.2.3

### DISCUSSION THREAD

**Discussion by Soothfast** [xournalpp/xournalpp#5556] [Q&A]
Version 1.2.3 of xournalpp leaves me with no template file.  It is gone.  I never had to configure it when I started out with version 1.2.1, since the template was just there automatically.  

I cannot find a template file anywhere.  It must exist, surely.  But where in Ubuntu, using TexStudio?  Under usr --> share there is no xournalpp folder, so that's out.  The template folder inside the TexStudio folder is empty.

EDIT:  I can see where I might theoretically place a template file in the TexStudio folder, but I don't know if that's the best place.  If it's an acceptable place, then I'm just left with two questions:  

(1) What happened to the template file I used to have?
(2) What exactly goes into making a template file, anyway?

FINAL EDIT:  I found default_template.tex on my Windows laptop, and put it someplace on my Ubuntu system.  Now Latex seems to work again.  When I search for "default_template" on the Ubuntu system, only the file I imported from the laptop shows up.  Question (1) above still stands.

**Comment by rolandlo** [comment]
The `default_template.tex` resides in the folder `/snap/xournalpp/current/usr/share/xournalpp/resources/`. You can put a copy into your home folder somewhere. Probably the template folder inside the TexStudio folder is fine.

For the snap package `/snap/xournalpp/current` is a symlink to `/snap/xournalpp/<current version number>`. The filename for the template in use is saved with that version number in its path, so it breaks on update. You avoid that problem by using a copy in your home folder instead.

By the way the template file can also be downloaded from the [repository](https://raw.githubusercontent.com/xournalpp/xournalpp/master/resources/default_template.tex).

For adding stuff to your template file see https://xournalpp.github.io/guide/tools/latex/#template

**Reply by Soothfast** [reply]
As ever, thanks for taking time to reply.  

My problem with the path you give to find default_template.tex is that the usr/share/xournalpp/resources/ part of it does not exist for me.  That is, there is no "usr" folder inside the "current" folder.  There are no hidden files -- I checked using Ctrl+h.

The problem of nonexistent folders is cropping up as a running theme in my battles with xournalpp.  Needless to say, this renders most of the tutorials on this site useless to me.   I have not modified, moved, or deleted folders since installing the program last fall, so I have no explanation for this.

But yes, when I recovered the default template, I kept it in the TexStudio folder.   I should hope that will prevent future updates from breaking the path.
